### ISSUE TITLE
controlnet训练

### ISSUE THREAD

**Issue by lsx193201** [kohya-ss/sd-scripts#1956]
我使用80G显存云服务器上使用https://github.com/huggingface/diffusers的项目训练17688数据量的controlnet模型，使用的训练命令如下：
accelerate launch /root/workspace/diffusers/examples/controlnet/train_controlnet_flux.py --pretrained_model_name_or_path="/root/workspace/cs4fc47hri0c73d6rs50/FLUX.1-dev" --jsonl_for_train "/root/workspace/cs4fc47hri0c73d6rs50/dataset/damodel_flux_cn.jsonl" --conditioning_image_column=conditioning_image --image_column=image --caption_column=text --output_dir="ComfyUI/models/controlnet/flux_cn_512_10000" --mixed_precision="bf16" --resolution=512 --learning_rate=1e-5 --max_train_steps=10000 --checkpointing_steps=500 --train_batch_size=1 --gradient_accumulation_steps=2 --num_double_layers=4 --num_single_layers=0 --seed=42 --save_weight_dtype="bf16" --use_adafactor --validation_steps=250 --validation_image "/root/workspace/diffusers/examples/controlnet/test.png" --validation_prompt "A cute, fluffy dog with a straw hat adorned with flowers, sitting on a table, beach scene in the background, ocean waves, palm leaves, and a tropical drink with an umbrella and lime." --num_validation_images=1 --report_to="tensorboard" 
最后模型的效果可以满足我的对图片提线的需求。
我在24G显存的云服务器上使用该项目的训练代码训练17688数据量的controlnet模型，得到的模型训练效果一直不好，不能对图片完成提线操作。我一共经过了三次训练，三次训练的详细细节如下：
测试图片
![Image](https://github.com/user-attachments/assets/7c2cfcf9-1dfc-4787-b4d1-e6c9b2bc27e9)
**第一次训练**：
训练分辨率512
训练批次4
训练步数1180
学习率2e-5
优化器adamw8bit
调度器默认
梯度累计1
每一百步一测试
交换块16
训练命令如下：
accelerate launch --mixed_precision bf16 --num_cpu_threads_per_process 1 autodl-tmp/kohya/sd-scripts-sd3/flux_train_control_net.py --pretrained_model_name_or_path autodl-tmp/stable-diffusion-webui-forge/models/Stable-diffusion/flux1-dev.safetensors --clip_l autodl-tmp/stable-diffusion-webui-forge/models/text_encoder/clip_l.safetensors --t5xxl autodl-tmp/stable-diffusion-webui-forge/models/text_encoder/t5xxl_fp16.safetensors --ae autodl-tmp/stable-diffusion-webui-forge/models/VAE/ae.safetensors --save_model_as safetensors --sdpa --persistent_data_loader_workers --max_data_loader_n_workers 1 --seed 42125 --gradient_checkpointing --mixed_precision bf16 --optimizer_type adamw8bit --learning_rate 2e-5 --highvram --save_every_n_steps 100 --dataset_config autodl-tmp/kohya/sd-scripts-sd3/flux_cn_512.toml --output_dir autodl-tmp/ComfyUI/models/controlnet/flux_cn_17688_512_4 --output_name flux_cn_17688_512_4 --timestep_sampling shift --discrete_flow_shift 3.1582 --model_prediction_type raw --guidance_scale 1.0 --logging_dir="autodl-tmp/kohya/logs/cn"  --gradient_accumulation_steps 1 --max_train_steps 1180 --sample_every_n_steps  100--sample_prompts autodl-tmp/kohya/dataset/test/prompts.toml --lr_scheduler constant  --blocks_to_swap 16 --cache_latents_to_disk --cache_text_encoder_outputs_to_disk
最后得到测试图效果如下：

![Image](https://github.com/user-attachments/assets/ce5ff278-8f2e-4240-97b9-0f60609046e8)
**第二次训练**：
训练分辨率512
训练批次3
梯度累计8
训练步数2400
学习率2e-5
优化器adamw8bit
调度器linear
每一百步一测试
交换块16
训练命令如下：
accelerate launch --mixed_precision bf16 --num_cpu_threads_per_process 1 autodl-tmp/kohya/sd-scripts-sd3/flux_train_control_net.py --pretrained_model_name_or_path autodl-tmp/stable-diffusion-webui-forge/models/Stable-diffusion/flux1-dev.safetensors --clip_l autodl-tmp/stable-diffusion-webui-forge/models/text_encoder/clip_l.safetensors --t5xxl autodl-tmp/stable-diffusion-webui-forge/models/text_encoder/t5xxl_fp16.safetensors --ae autodl-tmp/stable-diffusion-webui-forge/models/VAE/ae.safetensors --save_model_as safetensors --sdpa --persistent_data_loader_workers --max_data_loader_n_workers 1 --seed 42125 --gradient_checkpointing --mixed_precision bf16 --optimizer_type adamw8bit --learning_rate 2e-5 --highvram --save_every_n_steps 100--dataset_config autodl-tmp/kohya/sd-scripts-sd3/flux_cn_512.toml --output_dir autodl-tmp/ComfyUI/models/controlnet/flux_cn_17688_512_4 --output_name flux_cn_17688_512_4 --timestep_sampling shift --discrete_flow_shift 3.1582 --model_prediction_type raw --guidance_scale 1.0 --logging_dir="autodl-tmp/kohya/logs/cn"  --gradient_accumulation_steps 8 --max_train_steps 2400 --lr_scheduler linear --sample_every_n_steps  100--sample_prompts autodl-tmp/kohya/dataset/test/prompts.toml --lr_scheduler constant  --blocks_to_swap 16 --cache_latents_to_disk --cache_text_encoder_outputs_to_disk

![Image](https://github.com/user-attachments/assets/e7cf5d42-48cc-4ba5-91bf-771f7287cc25)
**第三次训练**
训练分辨率512
训练批次1
梯度累计2
训练步数30000
学习率1e-5
优化器adafactor
调度器constant
每一百步一测试
交换块10
accelerate launch --mixed_precision bf16 --num_cpu_threads_per_process 1 autodl-tmp/kohya/sd-scripts-sd3/flux_train_control_net.py --pretrained_model_name_or_path autodl-tmp/stable-diffusion-webui-forge/models/Stable-diffusion/flux1-dev.safetensors --clip_l autodl-tmp/stable-diffusion-webui-forge/models/text_encoder/clip_l.safetensors --t5xxl autodl-tmp/stable-diffusion-webui-forge/models/text_encoder/t5xxl_fp16.safetensors --ae autodl-tmp/stable-diffusion-webui-forge/models/VAE/ae.safetensors --save_model_as safetensors --sdpa --persistent_data_loader_workers --max_data_loader_n_workers 1 --seed 42125 --gradient_checkpointing --mixed_precision bf16 --optimizer_type adafactor --learning_rate 1e-5 --highvram --save_every_n_steps 1500 --dataset_config autodl-tmp/kohya/sd-scripts-sd3/flux_cn_512.toml --output_dir autodl-tmp/ComfyUI/models/controlnet/flux_cn_17688_512_4 --output_name flux_cn_17688_512_4 --timestep_sampling shift --discrete_flow_shift 3.1582 --model_prediction_type raw --guidance_scale 1.0 --logging_dir="autodl-tmp/kohya/logs/cn"  --gradient_accumulation_steps 2 --max_train_steps 30000 --sample_every_n_steps  100 --sample_prompts autodl-tmp/kohya/dataset/test/prompts.toml --lr_scheduler constant  --blocks_to_swap 10 --cache_latents_to_disk --cache_text_encoder_outputs_to_disk 
![Image](https://github.com/user-attachments/assets/35e43130-2c13-4e45-850a-5777f1b62242)
最后一次训练得到的模型效果可以大概提线我的测试图，但是他会出现很多的颜色，而且在24GB显存和80GB显存训练时损失都出现了震荡，有完成正常训练controlnet模型能够给我提供些指导吗
